horovod/torch/optimizer.py

```diff
@@ -101,6 +106,29 @@ def set_backward_passes_per_step(self, passes):
             self._allreduce_delay[p] = self.backward_passes_per_step
 
     def _register_hooks(self):
+
+        if self._num_groups > 0:
+            p_list = []
+            # Get list of parameters with grads
+            for param_group in self.param_groups:
+                for p in param_group['params']:
+                    if p.requires_grad:
+                        p_list.append(p)
+
+            # To ensure parameter order and group formation is consistent, broadcast p_list order
+            # from rank 0 and use for every worker
+            p_list_names = [self._parameter_names.get(p) for p in p_list]
+            p_list_names = broadcast_object(p_list_names, root_rank=0)
```

Comment:
This helper function has turned out to be really useful.

Reply:
Yes, I agree. 👍